docs/architecture/adr-002-globalfee.md

```diff
@@ -0,0 +1,149 @@
+# ADR 002: Globalfee Module
+
+## Changelog
+* 2023-06-12: Initial Draft
+
+## Status
+Globalfee module is introduced into gaia from v8.0.0, and is refactored in v11.0.0.
+
+## Context
+
+The globalfee module was created to manage a parameter called `MinimumGasPricesParam`, which sets a network-wide minimum fee. The intention was to stop random denominations from entering fee collections and to reduce the time validators take to check a long list of transaction fees. However, the initial version of the globalfee module had some issues:
+
+- In the globalfee module, several SDK coins methods were redefined because of the allowance of zero-value coins in the `MinimumGasPricesParam`. The `MinimumGasPricesParam` is of `sdk.DecCoins` type. In the cosmos-sdk, `sdk.DecCoins` are [sanitized](https://github.com/cosmos/cosmos-sdk/blob/67f04e629623d4691c4b2e48806f7793a3aa211e/types/dec_coin.go#L160-L177) to remove zero-value coins. As a result, several methods from `sdk.Coins` were [redefined in the Gaia fee antehandler](https://github.com/cosmos/gaia/blob/890ab3aa2e5788537b0d2ebc9bafdc968340e0e5/x/globalfee/ante/fee_utils.go#L46-L104).
```

Comment:
nit: use SDK and comos-sdk consistently (either one or the other). Probably better to use Cosmos SDK.